pkg/controller/replicaset/replica_set.go

```diff
@@ -327,11 +327,6 @@ func (rsc *ReplicaSetController) updatePod(old, cur interface{}) {
 	curPod := cur.(*api.Pod)
 	oldPod := old.(*api.Pod)
 	glog.V(4).Infof("Pod %s updated %+v -> %+v.", curPod.Name, oldPod, curPod)
-	rs := rsc.getPodReplicaSet(curPod)
-	if rs == nil {
-		return
-	}
-
 	if curPod.DeletionTimestamp != nil {
```

Comment:
If deletion timestamp is set on the cur pod and old pod has a different rc (i.e we got a single update with deletion timestamp and label change -- this can happen if both updates hit during a relist), we should invoke deletePod with the old one too. It's safe to do this becase deletion timestamp can never be unset.